src/stencil.cpp

```diff
@@ -304,87 +303,107 @@ bool Stencil::traverse_dag(DAGNode* node)
     }
 
     // execute this node and return a future
-    //if (thisIndex.x == 0 && thisIndex.y == 0)
-    DEBUG_PRINT("(%i, %i)> Send ghosts called for %i\n", thisIndex.x, thisIndex.y, kernel_node->node_id);
-    send_ghost_data(kernel_node);
+    // if (thisIndex.x == 0 && thisIndex.y == 0)
+    bool ghost_exchange_needed = false;
+
+    for (int i = 0; i < kernel_node->inputs.size(); i++)
+    {
+        int ghost_depth = arrays[kernel_node->inputs[i]]->ghost_depth;
```

Comment:
This shouldn't use the array's ghost depth, but the `ghost_info` from `Kernel`. Here is an example,

```
@kernel
def foo(a, b):
    # ghost info for  this kernel says a: 0, b: 0

@kernel
def bar(a, b):
    # ghost info for  this kernel says a: 1, b: 1

x = create_array()
y = create_array()

foo(x, y)
bar(x, y)
```

In the above example  `x->ghost_depth` and `y->ghost_depth` is going to be 1 for both since those arrays are used to call `bar`. But the call to `foo` doesn't need ghost transfers because that operation doesn't access any ghost data